src/ui/component/selectAsset/view.jsx

```diff
@@ -0,0 +1,150 @@
+// @flow
+
+import React, { useState } from 'react';
+import { FormField } from 'component/common/form';
+import FileSelector from 'component/common/file-selector';
+import Button from 'component/button';
+import fs from 'fs';
+import path from 'path';
+
+const filters = [
+  {
+    name: __('Thumbnail Image'),
+    extensions: ['png', 'jpg', 'jpeg', 'gif'],
+  },
+];
+
+const SOURCE_URL = 'url';
+const SOURCE_UPLOAD = 'upload';
+const SPEECH_READY = 'READY';
+const SPEECH_UPLOADING = 'UPLOADING';
+type Props = {
+  assetName: string,
+  currentValue: ?string,
+  onUpdate: string => void,
+  recommended: string,
+};
+
+function SelectAsset(props: Props) {
+  const { onUpdate, assetName, currentValue, recommended } = props;
+  const [assetSource, setAssetSource] = useState(SOURCE_URL);
+  const [pathSelected, setPathSelected] = useState('');
+  const [uploadStatus, setUploadStatus] = useState(SPEECH_READY);
+
+  function doUploadAsset(filePath, thumbnailBuffer) {
+    let thumbnail, fileExt, fileName, fileType;
+    if (filePath) {
+      thumbnail = fs.readFileSync(filePath);
+      fileExt = path.extname(filePath);
+      fileName = path.basename(filePath);
+      fileType = `image/${fileExt.slice(1)}`;
+    } else if (thumbnailBuffer) {
+      thumbnail = thumbnailBuffer;
+      fileExt = '.png';
+      fileName = 'thumbnail.png';
+      fileType = 'image/png';
+    } else {
+      return null;
+    }
+
+    const makeid = () => {
+      let text = '';
+      const possible = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
+      for (let i = 0; i < 24; i += 1) text += possible.charAt(Math.floor(Math.random() * 62));
+      return text;
+    };
```

Comment:
We already use the `uuid` module, lets just use that here.

Reply:
flow doesn't like import {v4 as uuidv4} from uuid;